### ISSUE TITLE
Google Colaboratory session crashed when training on GPU with the same metric in eval_metric and custom_metric

### ISSUE THREAD

**Issue by ezavarygin** [catboost/catboost#932]
Problem: Google Colaboratory session is crashed when training on GPU with the same metric in `eval_metric` and `custom_metric`
catboost version: 0.16.1
Operating System: Ubuntu 18.04.2 LTS (Google Colaboratory)
GPU: Tesla K80 GPU (Google Colaboratory)

The Colaboratory runtime is crashed when `eval_metric` and `custom_metric` have the same metric. This happens when running on GPU. No problem arises when running on CPU. The following example reproduces the problem:

```
from catboost.datasets import amazon
from catboost import CatBoostClassifier
from sklearn.model_selection import train_test_split

amazon_train, amazon_test = amazon()

X_train, X_valid, y_train, y_valid = train_test_split(
    amazon_train.drop('ACTION', axis=1), 
    amazon_train.ACTION, 
    test_size=0.3, 
    random_state=4
)
cat_features = list(range(X_train.shape[1]))

model = CatBoostClassifier(
    learning_rate=0.1,
    iterations=20,
    eval_metric= 'AUC',
    loss_function='Logloss',
    custom_metric=['AUC', 'F1'],
    task_type='GPU'

)
model.fit(
    X_train, 
    y_train, 
    cat_features=cat_features, 
    eval_set=(X_valid, y_valid),
    verbose=True
)
```

The runtime crashes with the following message:

![Screen Shot 2019-08-01 at 10 54 06](https://user-images.githubusercontent.com/18361119/62279620-20f93900-b453-11e9-95dc-b9a4e5bafbf8.png)

The logs are:

![Screen Shot 2019-08-01 at 11 56 03](https://user-images.githubusercontent.com/18361119/62279762-63227a80-b453-11e9-9814-ab2e300f1706.png)

No crashing happens when I remove `"AUC"` from `custom_metric` or change `eval_metric` to something different from `"AUC"` and `"F1"`.  The problem is not specific to `"AUC"`, e.g. setting `eval_metric` to `"F1"` in the example above still leads to the crash.

**Comment by annaveronika** [comment]
We'll look into this, thanks for the report! I think it's the same problem as #931

**Comment by annaveronika** [comment]
The fix is on github, will be on pypi in our next release.
